include/swift/AST/LayoutConstraint.h

```diff
@@ -0,0 +1,107 @@
+//===-- LayoutConstraint.h - Layout constraints types and APIs --*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines types and APIs for layout constraints.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_LAYOUT_CONSTRAINT_H
+#define SWIFT_LAYOUT_CONSTRAINT_H
+
+#include "llvm/ADT/StringRef.h"
+
+namespace swift {
+
+class Type;
+class TypeRepr;
+
+/// Describes a layout constraint information.
+enum class LayoutConstraintKind {
+  // It is not a known layout constraint.
+  UnknownLayout,
+  Trivial8 = 1,
```

Comment:
I don't think this stuff needs to go through swift-evolution, it is best to keep it as an underscored attribute since it is closely tied to the optimizer implementation, it will give us freedom to change it in the future and avoid the inevitable bikeshedding and unreasonable feature requests :-)